openmls/src/group/core_group/test_core_group.rs

```diff
@@ -702,3 +702,199 @@ fn test_own_commit_processing(
+        Some(ratchet_tree),
+        bob_kpb,
+        backend,
+    )
+    .expect("Error joining group.");
+
+    // Alice adds Charlie and removes Bob in the same commit.
+    let bob_kp_ref = alice_group
+        .treesync()
+        .leaves()
+        .expect("error getting leaves")
+        .values()
+        .find(|&kp| kp.credential().identity() == b"Bob")
+        .expect("Couldn't find Bob in tree.")
+        .hash_ref(backend.crypto())
+        .expect("error computing hash ref");
+    let bob_remove_proposal = alice_group
+        .create_remove_proposal(
+            framing_parameters,
+            &alice_credential_bundle,
+            &bob_kp_ref,
+            backend,
+        )
+        .expect("Could not create proposal");
+
+    let charlie_add_proposal = alice_group
+        .create_add_proposal(
+            framing_parameters,
+            &alice_credential_bundle,
+            charlie_kpb.key_package().clone(),
+            backend,
+        )
+        .expect("Could not create proposal");
+
+    let mut remove_add_proposal_store = ProposalStore::from_queued_proposal(
+        QueuedProposal::from_mls_plaintext(ciphersuite, backend, bob_remove_proposal)
+            .expect("Could not create QueuedProposal."),
+    );
+
+    remove_add_proposal_store.add(
+        QueuedProposal::from_mls_plaintext(ciphersuite, backend, charlie_add_proposal)
+            .expect("Could not create QueuedProposal."),
+    );
+
+    let params = CreateCommitParams::builder()
+        .framing_parameters(framing_parameters)
+        .credential_bundle(&alice_credential_bundle)
+        .proposal_store(&remove_add_proposal_store)
+        .build();
+    let remove_add_commit_result = alice_group
+        .create_commit(params, backend)
+        .expect("Error creating commit");
+
+    let staged_commit = bob_group
+        .stage_commit(
+            &remove_add_commit_result.commit,
+            &remove_add_proposal_store,
+            &[],
+            backend,
+        )
+        .expect("error staging commit");
+    bob_group
+        .merge_commit(staged_commit)
+        .expect("error merging commit");
+
+    alice_group
+        .merge_commit(remove_add_commit_result.staged_commit)
+        .expect("error merging pending commit");
+
+    let ratchet_tree = alice_group.treesync().export_nodes();
+
+    let charlie_group = CoreGroup::new_from_welcome(
+        remove_add_commit_result
+            .welcome_option
+            .expect("An unexpected error occurred."),
+        Some(ratchet_tree),
+        charlie_kpb,
+        backend,
+    )
+    .expect("Error joining group.");
+
+    // We can now check that Bob correctly processed his and applied the changes
+    // to his tree after he was removed by comparing membership lists. In
+    // particular, Bob's list should show that he was removed and Charlie was
+    // added.
+    let alice_members: Vec<&Credential> = alice_group
+        .treesync()
```

Comment:
Why not use the `members()` function here?

Reply:
Because this is on the `core_group` level. The `members()` function is on the `mls_group` level.